collector/src/execute.rs

```diff
@@ -44,26 +58,48 @@ fn rename<P: AsRef<Path>, Q: AsRef<Path>>(from: P, to: Q) -> anyhow::Result<()>
     Ok(())
 }
 
-fn touch(root: &Path, path: &Path) -> anyhow::Result<()> {
-    let mut cmd = Command::new("touch");
-    cmd.current_dir(root).arg(path);
-    command_output(&mut cmd).with_context(|| format!("touching {:?} in {:?}", path, root))?;
+fn touch(path: &Path) -> anyhow::Result<()> {
+    log::info!("touching file {:?}", path);
+
+    filetime::set_file_mtime(path, filetime::FileTime::now()).with_context(|| format!("touching file {:?}", path))?;
+
     Ok(())
 }
 
 fn touch_all(path: &Path) -> anyhow::Result<()> {
-    let mut cmd = Command::new("bash");
-    // Don't touch files in `target/`, since they're likely generated by build scripts and might be from a dependency.
-    // Don't touch build scripts, which confuses the wrapped rustc.
-    cmd.current_dir(path)
-        .args(&["-c", "find . -path ./target -prune -false -o -name '*.rs' | grep -v '^./build.rs$' | xargs touch"]);
-    command_output(&mut cmd).with_context(|| format!("touching all .rs in {:?}", path))?;
-    // We also delete the cmake caches to avoid errors when moving directories around.
-    // This might be a bit slower but at least things build
-    let mut cmd = Command::new("bash");
-    cmd.current_dir(path)
-        .args(&["-c", "find . -name 'CMakeCache.txt' -delete"]);
-    command_output(&mut cmd).with_context(|| format!("deleting cmake caches in {:?}", path))?;
+    fn is_valid(path: &Path) -> bool {
+        let target_dir = Component::Normal(OsStr::new("target"));
+
+        // Don't touch files in `target/`, since they're likely generated by build scripts and might be from a dependency.
+        if path.components().any(|component| component == target_dir) {
```

Comment:
This would break if any subfolder is named `target` no?

Reply:
I think that's only true if the subfolder actually needs touching; I'm not sure that will happen in practice. Realistically we can likely get away with touching the crate "roots" only, but those are a bit annoying to identify.